File: c/ex6.c
/*
* program: ex6.c
* Exercise 6 from Learn C the Hard Way
*/

#include <stdio.h>

int main(int argc, char *argv[]) 
{
	int distance = 100;
	float power = 2.345f;
	double super_power = 56789.4532;
	char initial = 'J';
	char first_name[] = "Gav";
	char last_name[] = "Coombes";

	printf("You are %d miles away.\n", distance);
	printf("You have %f levels of power.\n", power);
	printf("You have %f awesome super powers.\n", super_power);
	printf("I have an initial %c.\n", initial);
	printf("I have a first name %s.\n", first_name);
	printf("I have a last name %s.\n", last_name);
	printf("My whole name is %s %c. %s.\n", 
		first_name, initial, last_name);

	return 0;
}

File: rust/ex6.rs
// ex6.rs
// Value literals

fn main() {
	
	let distance = 100i;
	let power = 2.345f32;
	let super_power = 56789.4532f64;
	let initial = 'J';
	let first_name = "Gav";
	let last_name = "Coombes";

	println!("You are {:d} miles away.", distance);
	println!("You have {:f} levels of power.", power);
	println!("You have {:f} awesome super powers.", super_power);
	println!("I have an initial {:c}.", initial);
	println!("I have a first name {:s}.", first_name);
	println!("I have a last name {:s}.", last_name);
	println!("My whole name is {:s} {:c}. {:s}.", 
		first_name, initial, last_name);
}



